File: output/ChatGPT/avatar/Python/C++/codeforces_373_B.cpp
#include <iostream>
using namespace std;

long long sum(long long k) {
    long long ret = 0;
    long long pw = 10;
    long long len = 1;
    while (true) {
        long long cur = min(pw - 1, k);
        long long prev = pw / 10;
        ret += (cur - prev + 1) * len;
        if (pw - 1 >= k) {
            break;
        }
        len += 1;
        pw *= 10;
    }
    return ret;
}

int main() {
    long long w, m, k;
    cin >> w >> m >> k;
    long long lo = 0;
    long long hi = 1e18;
    while (hi - lo > 1) {
        long long md = (lo + hi) / 2;
        long long c = sum(m + md - 1) - sum(m - 1);
        if (c * k <= w) {
            lo = md;
        }
        else {
            hi = md;
        }
    }
    cout << lo << endl;
    return 0;
}

File: output/CodeLlama/avatar/Python/Java/codeforces_373_B.java
public class codeforces_373_B {
    public static int sum(int k) {
        int ret = 0;
        int pw = 10;
        int len = 1;
        while (true) {
            int cur = Math.min(pw - 1, k);
            int prev = pw / 10;
            ret += (cur - prev + 1) * len;
            if (pw - 1 >= k) {
                break;
            }
            len += 1;
            pw *= 10;
        }
        return ret;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int w = sc.nextInt();
        int m = sc.nextInt();
        int k = sc.nextInt();
        int lo = 0;
        int hi = (int) 1e18;
        while (hi - lo > 1) {
            int md = (lo + hi) / 2;
            int c = sum(m + md - 1) - sum(m - 1);
            if (c * k <= w) {
                lo = md;
            } else {
                hi = md;
            }
        }
        System.out.println(lo);
    }
}